### PR TITLE
Cy 2268 handle newlines in keys

### PR DESCRIPTION
**Pull request by EarthmanT**

### CODE REVIEW

File: cloudify_gcp/utils.py [hunk 1]
```diff
@@ -349,14 +353,16 @@ def _get_gcp_config_from_properties():
...
-
         # If no network is specified, assume the GCP default network, 'default'
         gcp_config.setdefault('network', 'default')
-
         return update_zone(gcp_config)
     return gcp_config
 
 
+def handle_newlines_in_key_file(key_content):
+    return key_content.encode('utf-8').replace('\n', '\\n')
```
Comment (0lvin) [inline cloudify_gcp/utils.py]: Looks as not used.

Comment (EarthmanT) [reply cloudify_gcp/utils.py]: yep removed

Comment (mabuaisha) [reply cloudify_gcp/utils.py]: Is this used ?

File: cloudify_gcp/utils.py [hunk 2]
```diff
@@ -330,8 +330,12 @@ def _get_gcp_config_from_properties():
...
                     e,
                     ))
-    if not gcp_config.get('auth'):
-        raise NonRecoverableError("No auth provided")
+    if 'auth' not in gcp_config:
+        raise NonRecoverableError("No auth provided in gcp_config.")
+
+    if gcp_config['auth'].get('private_key'):
+        gcp_config['auth']['private_key'] = gcp_config['auth'][
+            'private_key'].replace('\\n', '\n')
```
Comment (mabuaisha) [inline cloudify_gcp/utils.py]: I think you should call the function you define below

**Pull request merged**